### PR TITLE
Jwineman/pi 701 constructor based dependencies

### PR DESCRIPTION
**Pull request by jwineman**

### CODE REVIEW

File: Controller/Adminhtml/Plugin/Proxy.php [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+use \CF\API\Plugin;
+use \CF\API\Request;
+use \CF\Integration\DefaultConfig;
+use \CF\Integration\DefaultIntegration;
+use \CF\Router\DefaultRestAPIRouter;
+use \CloudFlare\Plugin\Backend\ClientRoutes;
+use \CloudFlare\Plugin\Backend\DataStore;
+use \CloudFlare\Plugin\Backend\MagentoAPI;
+use \CloudFlare\Plugin\Backend\PluginRoutes;
+use \CloudFlare\Plugin\Model\KeyValueFactory;
```
Comment (thellimist) [inline Controller/Adminhtml/Plugin/Proxy.php]: Why is this imported?

File: Controller/Adminhtml/Plugin/Proxy.php [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+
 use \Magento\Backend\App\AbstractAction;
 use \Magento\Backend\App\Action\Context;
+use \Magento\Framework\App\DeploymentConfig\Reader;
```
Comment (thellimist) [inline Controller/Adminhtml/Plugin/Proxy.php]: Same with this

File: Controller/Adminhtml/Plugin/Proxy.php (hunk 1 above)
Comment (jwineman) [inline Controller/Adminhtml/Plugin/Proxy.php]: Just left over, I'll remove.

File: Controller/Adminhtml/Plugin/Proxy.php (hunk 2 above)
Comment (jwineman) [inline Controller/Adminhtml/Plugin/Proxy.php]: Left over, I'll remove.

**Pull request merged**